python/cudf/cudf/core/column/numerical.py

```diff
@@ -699,18 +699,25 @@ def digitize(column, bins, right=False):
     ----------
     column : Column
         Input column.
-    bins : np.array
-        1-D monotonically increasing array of bins with same type as `column`.
+    bins : Column-like
+        1-D column-like object of bins with same type as `column`, should be
+        monotonically increasing.
     right : bool
         Indicates whether interval contains the right or left bin edge.
 
     Returns
     -------
     A device array containing the indices
     """
-    assert column.dtype == bins.dtype
-    bins_buf = Buffer(bins.view("|u1"))
-    bin_col = NumericalColumn(data=bins_buf, dtype=bins.dtype)
+    if not column.dtype == bins.dtype:
+        raise ValueError(
+            "Digitize() expects bins and input column have the same datatype."
+        )
+
+    bin_col = as_column(bins, dtype=bins.dtype)
+    if bin_col.nullable:
+        raise ValueError("Argument `bins` contains null entries.")
```

Comment:
This is probably a matter of opinion, but I think the following is a bit better:

```suggestion
        raise ValueError("`bins` cannot contain null entries.")
```

Feel free to ignore :)